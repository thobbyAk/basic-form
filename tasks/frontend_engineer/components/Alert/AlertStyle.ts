import styled from "styled-components";

export const AlertContainer = styled.div`
  display: none;
  align-items: center;
  justify-content: center;
  width: 100%;
  padding: 10;
  background-color: ${({ theme }) => theme.colors.alert};
`;
