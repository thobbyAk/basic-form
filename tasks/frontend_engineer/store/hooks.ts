import { useDispatch,useSelector, TypedUseSelectorHook } from "react-redux";
import { AppDispatch } from ".";
import { IRootState } from "@/reducer";

export const useAppDispatch = () => {
    const dispatch = useDispatch<AppDispatch>();
  
    return dispatch;
  };

  export const useAppStateSelector: TypedUseSelectorHook<IRootState> =
  useSelector;