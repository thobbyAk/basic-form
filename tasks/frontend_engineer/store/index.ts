import rootReducer from '@/reducer';
import { configureStore } from '@reduxjs/toolkit';

export const store = configureStore({
    reducer: rootReducer
});

export type AppDispatch = typeof store.dispatch
export default store;