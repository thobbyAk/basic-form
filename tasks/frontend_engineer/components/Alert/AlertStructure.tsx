import React, { FC } from 'react'
import { AlertContainer } from "./AlertStyle";

const AlertStructure: FC<{ message: string }> = ({ message }) => {
  return (
    <AlertContainer>
      <h1>{message}</h1>
    </AlertContainer>
  );
}

export default AlertStructure;