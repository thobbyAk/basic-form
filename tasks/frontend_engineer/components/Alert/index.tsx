import React from "react";
import AlertComponent from "./AlertStructure";
import { useAppStateSelector } from "@/store/hooks";

const Alert = () => {
	const { message, showAlert } = useAppStateSelector(
		(state) => state.alertState
	);

	const componentProps = {
		message,
	};

	return (
		<React.Fragment>
			{showAlert && <AlertComponent {...componentProps} />}
		</React.Fragment>
	);
};

export default Alert;
