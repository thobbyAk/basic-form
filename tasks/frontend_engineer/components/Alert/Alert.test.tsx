import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Alert from ".";
import { Provider } from "react-redux";
import store from "@/store";

test("it should match the snapshot and render", async () => {
	const { container } = render(
		<Provider store={store}>
			<Alert />
		</Provider>
	);
	expect(container).toMatchSnapshot();
	expect(container).toBeInTheDocument();
});
