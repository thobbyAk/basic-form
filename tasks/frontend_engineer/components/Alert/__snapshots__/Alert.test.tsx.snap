// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match the snapshot and render 1`] = `<div />`;
